scanner/decisions/0009-nvd-cvss-scores.md

```diff
@@ -0,0 +1,148 @@
+Note: NVD tracks CVEs, so all of tis data is CVE-based.
+
+Currently, Scanner V4 provides a single CVSS score and severity, which is preferably from the vendor.
+In reality, the CVSS scores that are shown are from NVD except in the following cases:
+
+* RHEL-base images
+  * Red Hat-provided CVSS scores/severities are used.
+* OSV-provided data
+  * If OSV.dev has a CVSS score, then that score is used.
+    * The severity is derived from the score based on the CVSS version's specification.
+  * Note: From personal experience, it is very common for the score displayed by OSV is the same as NVD.
+
+Currently, there is only a need to have CVSS scores from the vendor (if available) and NVD (if available), no more.
+However, it is possible this requirement changes in the future, and we may want to support more than just two
+sources for vulnerability data.
+
+The focus is on CVEs and RHSAs, RHEAs, and RHBAs. This document will not consider other types of advisories such as
+Ubuntu's USNs.
+
+The current API looks like the following:
+
+```
+message VulnerabilityReport {
+  message Vulnerability {
+    enum Severity {
+      SEVERITY_UNSPECIFIED = 0;
+      SEVERITY_LOW = 1;
+      SEVERITY_MODERATE = 2;
+      SEVERITY_IMPORTANT = 3;
+      SEVERITY_CRITICAL = 4;
+    }
+    message CVSS {
+      message V2 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      V2 v2 = 1;
+      V3 v3 = 2;
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+  }
+  ...
+}
+```
+
+## Decision
+
+The VulnerabilityReport API will be extended as follows:
+
+```
+message VulnerabilityReport {
+  message Vulnerability {
+    enum Severity {
+      SEVERITY_UNSPECIFIED = 0;
+      SEVERITY_LOW = 1;
+      SEVERITY_MODERATE = 2;
+      SEVERITY_IMPORTANT = 3;
+      SEVERITY_CRITICAL = 4;
+    }
+    message CVSS {
+      enum Source {
+        SOURCE_UNKNOWN = 0;
+        SOURCE_RED_HAT = 1;
+        SOURCE_OSV = 2;
+        SOURCE_NVD = 3;
+      }
+      message V2 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      Source source = 3; <-- New field.
+      V2 v2 = 1;
+      V3 v3 = 2;
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+    repeated CVSS cvss_metrics = 13; <-- New field.
+  }
+  ...
+}
+```
+
+There will be a new type plus two new fields added:
+
+* `CVSS.Source`
+  * This specifies the supported CVSS metrics data sources.
+* `CVSS.source`
+  * This specifies the source of the particular CVSS metrics.
+* `cvss_metrics`
+  * This is a list of each unique CVSS metric based on the source.
+
+The original `cvss` field will remain and will continue to represent the Scanner's preferred CVSS score.
+This is currently the score from the vulnerability's original data source, if available, otherwise NVD.
```

Comment:
```suggestion
The original `cvss` was replaced with `repeated CVSS cvss`, which contains all known CVSS for the vulnerability.
```

Then you can leave to Central to decide which CVSS to prefer and to show?

Reply:
From what I can tell, we plan on splitting things into "StackRox-preferred" and "NVD", so perhaps keeping this field as-is makes the most sense. Can you comment on this @daynewlee?

Reply:
@RTann  Yes, I didn't apply Victor's suggestions as we need to keep the **_cvss_** field and add **_cvss_metrics_** field